tools/mason/MasonNew.chpl

```diff
@@ -19,72 +19,74 @@
  */
 
 
-use Path;
+use FileSystem;
 use IO;
-use Spawn;
+use MasonArgParse;
+use MasonEnv;
+use MasonHelp;
 use MasonModify;
-use FileSystem;
-use MasonUtils;
 use MasonUpdate;
-use MasonHelp;
-use MasonEnv;
+use MasonUtils;
+use Path;
+use Spawn;
+
 
 /*
   Creates a new library project at a given directory
   mason new <projectName/directoryName>
 */
 proc masonNew(args: [] string) throws {
-  var vcs = true;
-  var show = false;
+  var parser = new argumentParser();
+  var helpFlag = parser.addFlag("help",
+                                opts=["-h","--help"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var showFlag = parser.addFlag(name="show",
+                                opts=["--show"],
+                                defaultValue=false,
+                                flagInversion=false);
+  var vcsFlag = parser.addFlag(name="vcs",
+                               opts=["--no-vcs"],
+                               defaultValue=false,
+                               flagInversion=false);
+  var nameOpt = parser.addOption(name="legalname",
```

Comment:
I've seen this `name="legalname"` pattern in a few cases and it appears that the `name` argument here isn't really used anywhere? Do you think it'd be reasonable for the argument parser to compute a unique name for each argument (e.g. if name isn't specified, it uses `"arg1"` `"arg2"` etc)?

Reply:
> Do you think it'd be reasonable for the argument parser to compute a unique name for each argument (e.g. if name isn't specified, it uses "arg1" "arg2" etc)?

possibly, but I think this relates to the [my other comment on this PR](https://github.com/chapel-lang/chapel/pull/18317#issuecomment-909445796). Although there may be a case to be made for allowing no name value, the name field will be used and should probably not be generic in most cases.